src/placeos-models/edge.cr

```diff
@@ -29,10 +33,12 @@ module PlaceOS::Model
       foreign_key: "edge_id",
     )
 
-    before_save :encrypt!
+    def valid?(test : String) : Bool
+      Encryption.decrypt(string: secret, id: salt, level: ENCRYPTION_LEVEL) == test
+    end
 
     def encrypt
```

Comment:
It looks like this encrypts that secret only. To make this a little clearer (particularly if there's a need to encrypt multiple attributes down the track, what about `encrypt_secret` to make this clearer? Ditto for the decrypt pair.

Reply:
`valid_secret?` might be clearer?

Reply:
oh you mean specifically regarding the attribute. yep all for that.

Reply:
Yep, sorry - the comment attached to the incorrect line.